### PR TITLE
fix(client): GA to GTM Migration

### PR DESCRIPTION
**Pull request by ahmaxed**
- feat/initial-setup
- feat: clean up calling analytics on each page
- feat: remove a/b testing logic
- feat: remove react-ga
- Removing Map related legacy events
- feat: standardize events
- feat: standarize events after rebase

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/redux/selectors.js [hunk 1]
```diff
@@ -13,13 +12,7 @@ export const currentChallengeIdSelector = state =>
...
 
 export const emailSelector = state => userSelector(state).email;
-export const isVariantASelector = state => {
-  const email = emailSelector(state);
-  // if the user is not signed in and the user info is not available.
-  // always return A the control variant
-  if (!email) return true;
-  return emailToABVariant(email).isVariantA;
-};
+
```
Comment (Sboonny) [inline client/src/redux/selectors.js:15]:
```suggestion
```

File: client/src/analytics/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import TagManager from 'react-gtm-module';
+
+import {
+  devAnalyticsId,
+  prodAnalyticsId
+} from '../../../config/analytics-settings';
+
+import envData from '../../../config/env.json';
+
+const { deploymentEnv } = envData;
```
Comment (Sboonny) [inline client/src/analytics/index.ts:10]:
```suggestion
import TagManager from 'react-gtm-module';
import {
  devAnalyticsId,
  prodAnalyticsId
} from '../../../config/analytics-settings';
import { deploymentEnv } from '../../../config/env.json';
```

File: client/src/analytics/index.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const gtmId = deploymentEnv === 'staging' ? devAnalyticsId : prodAnalyticsId;
+
+if (typeof document !== `undefined`) {
+  TagManager.initialize({ gtmId });
+}
```
Comment (Sboonny) [inline client/src/analytics/index.ts:16]:
```suggestion
TagManager.initialize({ gtmId });
```

Just wondering, is it possible for the document API to be `undefined`?

**Review by Sboonny — COMMENTED** [review]
Just small cleaning 👍

File: client/src/analytics/index.ts (hunk 3 above)
Comment (raisedadead) [reply client/src/analytics/index.ts:16]: Yes — it is possible. When not in a browser context. This check should be there.

**Pull request merged**